### PR TITLE
Make clear notifications work with threads

### PR DESCRIPTION
**Pull request by gsouquet**
Fixes https://github.com/vector-im/element-web/issues/23751
Works off https://github.com/matrix-org/matrix-js-sdk/pull/2878

## Checklist

* [x] Tests written for new code (and old code if feasible)
* [x] Linter and other CI checks pass
* [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## ✨ Features
 * Make clear notifications work with threads ([\#9575](https://github.com/matrix-org/matrix-react-sdk/pull/9575)). Fixes vector-im/element-web#23751.

### CODE REVIEW

File: src/utils/notifications.ts [hunk 1]
```diff
@@ -56,3 +58,28 @@ export function localNotificationsAreSilenced(cli: MatrixClient): boolean {
...
+
+            const lastRoomEvent = roomEvents?.[roomEvents?.length - 1];
+            const lastThreadLastEvent = lastThreadEvents?.[lastThreadEvents?.length - 1];
+
+            const lastEvent = (lastRoomEvent?.getTs() ?? 0) > (lastThreadLastEvent?.getTs() ?? 0)
+                ? lastRoomEvent
+                : lastThreadLastEvent;
+
+            if (lastEvent) {
+                const promise = client.sendReadReceipt(lastEvent, ReceiptType.Read, true);
```
Comment (SimonBrandner) [inline src/utils/notifications.ts:76]: What about private read receipts?

**Review by SimonBrandner — CHANGES_REQUESTED** [review]
Otherwise LGTM, it's just quite concerning we could leak public RRs here

File: src/utils/notifications.ts (hunk 1 above)
Comment (gsouquet) [reply src/utils/notifications.ts]: @SimonBrandner I am unsure how private read receipts are supposed to work. 

I have copied the logic that is in `TimelinePanel` that checks the `sendReadReceipts` setting per room to decide to send `ReceiptType.Read` if `true` and `ReceiptType.ReadPrivate` if `false`. 

Can you confirm this is correct? 
By the way, the clear notification feature did not support private read receipts before this pull request.

File: test/components/views/settings/Notifications-test.tsx [hunk 2]
```diff
@@ -300,4 +296,32 @@ describe('<Notifications />', () => {
...
+            room.addLiveEvents([message]);
+            room.setUnreadNotificationCount(NotificationCountType.Total, 1);
+
+            const { container } = await getComponentAndWait();
+            const clearNotificationEl = getByTestId(container, "clear-notifications");
+
+            fireEvent.click(clearNotificationEl);
+
+            expect(clearNotificationEl.className).toContain("mx_AccessibleButton_disabled");
+            expect(mockClient.sendReadReceipt).toHaveBeenCalled();
```
Comment (SimonBrandner) [inline test/components/views/settings/Notifications-test.tsx:320]: Could we check the args here?

**Review by SimonBrandner — APPROVED** [review]
LGTM now!

**Pull request merged**